Sources/SafeDICore/Generators/DependencyTreeGenerator.swift

```diff
@@ -389,16 +433,17 @@ public final class DependencyTreeGenerator {
 
             for childPropertyToGenerate in scope.propertiesToGenerate {
                 switch childPropertyToGenerate {
-                case let .instantiated(property, childScope, _):
-                    guard !childPropertyStack.contains(property) else {
+                case let .instantiated(childProperty, childScope, _):
+                    guard !childPropertyStack.contains(childProperty) else {
                         // There is a cycle in our scope tree. Do not re-enter it.
                         continue
                     }
-                    try validateReceivedProperties(
+                    try validatePropertiesAreFulfillable(
                         on: childScope,
                         receivableProperties: receivableProperties
-                            .union(scope.properties),
-                        property: property,
+                            .union(scope.properties)
+                            .removing(childProperty),
```

Comment:
brought this line back from #95. Now that we're doing our cycle checking before we're doing our "initialized in chain" checking, we can make this more semantically correct